cmake/toolchain/espressif/target.cmake

```
@@ -1,3 +1,21 @@
 # SPDX-License-Identifier: Apache-2.0
 
-# This file intentionally left blank.
+set_ifndef(ESPRESSIF_TOOLCHAIN_PATH "$ENV{ESPRESSIF_TOOLCHAIN_PATH}")
+set(       ESPRESSIF_TOOLCHAIN_PATH ${ESPRESSIF_TOOLCHAIN_PATH} CACHE PATH "")
+assert(    ESPRESSIF_TOOLCHAIN_PATH "ESPRESSIF_TOOLCHAIN_PATH is not set")
+
+set(TOOLCHAIN_HOME ${ESPRESSIF_TOOLCHAIN_PATH}/${SOC_TOOLCHAIN_NAME})
+
+set(COMPILER gcc)
+set(LINKER ld)
+set(BINTOOLS gnu)
+
+set(CROSS_COMPILE_TARGET ${SOC_TOOLCHAIN_NAME})
```

A Kconfig setting must be prefixed with `CONFIG_` when being used in CMake, but why not do like this:
```suggestion
set(CROSS_COMPILE_TARGET_xtensa_esp32    xtensa-esp32-elf)
set(CROSS_COMPILE_TARGET_xtensa_esp32s2  xtensa-esp32s2-elf)
set(CROSS_COMPILE_TARGET_riscv_esp32c3   riscv32-esp-elf)

set(CROSS_COMPILE_TARGET ${CROSS_COMPILE_TARGET_${ARCH}_${CONFIG_SOC}})
set(SYSROOT_TARGET       ${CROSS_COMPILE_TARGET})
```

That will keep the relation between the toolchain and the soc in same place as is done for the Zephyr SDK, and not start to move such knowledge into Kconfig.

---

@tejlmand thanks for the input, your suggestion is just pretty much better.

---

Thanks for the update.

Code wise i'm ready for +1, just a small observation I would like some feedback on.